src/openrct2/FileClassifier.cpp

```diff
@@ -95,57 +95,68 @@ static bool TryClassifyAsS6(IStream * stream, ClassifiedFileInfo * result)
 
 static bool TryClassifyAsS4(IStream * stream, ClassifiedFileInfo * result)
 {
+    bool success = false;
     uint64 originalPosition = stream->GetPosition();
-    size_t dataLength = (size_t)stream->GetLength();
-    uint8 * data = stream->ReadArray<uint8>(dataLength);
-    stream->SetPosition(originalPosition);
-    sint32 fileTypeVersion = sawyercoding_detect_file_type(data, dataLength);
-    Memory::Free(data);
+    try
+    {
+        size_t dataLength = (size_t)stream->GetLength();
+        std::unique_ptr<uint8> data(stream->ReadArray<uint8>(dataLength));
+        stream->SetPosition(originalPosition);
+        sint32 fileTypeVersion = sawyercoding_detect_file_type(data.get(), dataLength);
 
-    sint32 type = fileTypeVersion & FILE_TYPE_MASK;
-    sint32 version = fileTypeVersion & FILE_VERSION_MASK;
+        sint32 type = fileTypeVersion & FILE_TYPE_MASK;
+        sint32 version = fileTypeVersion & FILE_VERSION_MASK;
 
-    if (type == FILE_TYPE_SV4)
-    {
-        result->Type = FILE_TYPE::SAVED_GAME;
-        result->Version = version;
-        return true;
+        if (type == FILE_TYPE_SV4)
+        {
+            result->Type = FILE_TYPE::SAVED_GAME;
+            result->Version = version;
+            success = true;
+        }
+        else if (type == FILE_TYPE_SC4)
+        {
+            result->Type = FILE_TYPE::SCENARIO;
+            result->Version = version;
+            success = true;
+        }
     }
-    else if (type == FILE_TYPE_SC4)
+    catch (Exception)
```

Comment:
We should at very least log the error message (as in the other handler you added).